### PR TITLE
handheld-daemon: 3.4.1 -> 3.6.1

### PR DESCRIPTION
**Pull request by toast003**
Also install hhd's hwdb file, since it is needed for buttons to work properly on the MSI Claw, among other things

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by getchoo — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359282`

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>handheld-daemon</li>
    <li>handheld-daemon.dist</li>
  </ul>
</details>

File: pkgs/by-name/ha/handheld-daemon/package.nix [hunk 1]
```diff
@@ -5,17 +5,18 @@
   lib,
   python3,
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/handheld-daemon/package.nix:6]:
```suggestion
  python3Packages,
```

File: pkgs/by-name/ha/handheld-daemon/package.nix [hunk 2]
```diff
@@ -5,17 +5,18 @@
...
   toybox,
+  lsof
 }:
 python3.pkgs.buildPythonApplication rec {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/handheld-daemon/package.nix:10]:
```suggestion
python3Packages.buildPythonApplication rec {
```

File: pkgs/by-name/ha/handheld-daemon/package.nix [hunk 3]
```diff
@@ -5,17 +5,18 @@
...
 
   src = fetchFromGitHub {
     owner = "hhd-dev";
     repo = "hhd";
     rev = "refs/tags/v${version}";
-    hash = "sha256-hup9G4pTlTl68zLzhaDmH7nV8fp/I94Nuis68B8K2wY=";
+    hash = "sha256-IdpSRb66G+WzTv/BC29r2OjO1b4VdWbV6OSzOoiFAO0=";
   };
 
   propagatedBuildInputs = with python3.pkgs; [
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/handheld-daemon/package.nix:22]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/by-name/ha/handheld-daemon/package.nix [hunk 4]
```diff
@@ -46,6 +49,7 @@ python3.pkgs.buildPythonApplication rec {
 
   postInstall = ''
     install -Dm644 $src/usr/lib/udev/rules.d/83-hhd.rules -t $out/lib/udev/rules.d/
+    install -Dm644 $src/usr/lib/udev/hwdb.d/83-hhd.hwdb -t $out/lib/udev/hwdb.d/
   '';
 
   meta = with lib; {
```
Comment (GaetanLepage) [inline pkgs/by-name/ha/handheld-daemon/package.nix:55]:
```suggestion
  meta = {
```

**Review by GaetanLepage — CHANGES_REQUESTED** [review]

**Comment by getchoo** [comment]
@GaetanLepage: None of these suggestions are for changes introduced by this PR. They would be nice to have definitely, but we shouldn't be blocking on this

**Comment by GaetanLepage** [comment]
> @GaetanLepage: None of these suggestions are for changes introduced by this PR. They would be nice to have definitely, but we shouldn't be blocking on this

No definitely. Usually, I think that it is good to "clean" stuff when updating packages. I won't block your PR for this though.

**Review by GaetanLepage — APPROVED** [review]

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 359282`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>handheld-daemon</li>
    <li>handheld-daemon.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>handheld-daemon</li>
    <li>handheld-daemon.dist</li>
  </ul>
</details>

**Pull request merged**